File: ports/php/run_parser7.php
<?php
declare(strict_types=1);

require __DIR__ . '/ParserV7.php';

$grammar = __DIR__ . '/resources/grammar.json';
$lexicon = __DIR__ . '/resources/lexicon.json';
$beam = 8;

$args = $argv;
for ($i = 1; $i < count($args); $i++) {
    if ($args[$i] === '--grammar' && isset($args[$i+1])) $grammar = $args[++$i];
    elseif ($args[$i] === '--lexicon' && isset($args[$i+1])) $lexicon = $args[++$i];
    elseif ($args[$i] === '--beam' && isset($args[$i+1])) $beam = (int)$args[++$i];
}

$p = new ParserV7($grammar, $lexicon, $beam);

fwrite(STDOUT, "Ready. Pegá 1 oración por línea (CTRL+D para salir).\n");

while (($line = fgets(STDIN)) !== false) {
    $line = trim($line);
    if ($line === '') continue;

    $tree = $p->parseSentence($line);
    if ($tree === '') fwrite(STDOUT, "(NO-PARSE)\n");
    else fwrite(STDOUT, $tree . "\n");
}

File: ports/perl/run_parser7.pl
#!/usr/bin/env perl
use strict;
use warnings;
use utf8;

use FindBin qw($Bin);
use lib $Bin;

use ParserV7;

binmode(STDIN,  ":encoding(UTF-8)");
binmode(STDOUT, ":encoding(UTF-8)");
binmode(STDERR, ":encoding(UTF-8)");

my $grammar = "$Bin/resources/grammar.json";
my $lexicon = "$Bin/resources/lexicon.json";
my $beam    = 8;

# args: --grammar path --lexicon path --beam N
for (my $i=0; $i<@ARGV; $i++) {
    if ($ARGV[$i] eq '--grammar' && defined $ARGV[$i+1]) { $grammar = $ARGV[++$i]; next; }
    if ($ARGV[$i] eq '--lexicon' && defined $ARGV[$i+1]) { $lexicon = $ARGV[++$i]; next; }
    if ($ARGV[$i] eq '--beam'    && defined $ARGV[$i+1]) { $beam = 0 + $ARGV[++$i]; next; }
}

my $p = ParserV7->new(
    grammar_path => $grammar,
    lexicon_path => $lexicon,
    beam         => $beam,
);

print "Ready. Pegá 1 oración por línea (EOF para salir).\n";

while (defined(my $line = <STDIN>)) {
    chomp $line;
    next if $line =~ /^\s*$/;

    my $tree = $p->parse_sentence($line);
    if (!$tree) {
        print "(NO-PARSE)\n";
    } else {
        print $tree, "\n";
    }
}
